core/build.gradle.kts

```diff
@@ -15,22 +15,36 @@ kotlin {
     android {
         publishAllLibraryVariants()
     }
+    macosX64()
 
     sourceSets {
         val commonMain by getting {
             dependencies {
-                api(coroutines("core"))
+                api(coroutines("core", version = "1.3.9-native-mt-2"))
                 api(uuid())
             }
         }
 
+        val jsMain by getting {
+            dependencies {
+                implementation(coroutines("core"))
```

Comment:
How I understand this works (or at least hope; I'll see if I can dig a little to double check this):
The common code is pulling in `1.3.9-native-mt-2`, so all targets get that version, except in targets that we specify `1.4.1`, then Gradle knows that it can auto-bump to that version.
So targets that need `mt-2` use that version of Coroutines, the others that don't need that special branch of Coroutines we can tell Gradle to let those targets use `1.4.1`.

Reply:
Dependency insights:

<details>
<summary>Android</summary>

```
./gradlew :core:dependencyInsight --configuration androidReleaseCompileClasspath --dependency coroutines

> Task :core:dependencyInsight
org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1
   variant "apiElements" [
      org.gradle.category                 = library (not requested)
      org.gradle.dependency.bundling      = external (not requested)
      org.gradle.jvm.version              = 6 (not requested)
      org.gradle.libraryelements          = jar (not requested)
      org.gradle.usage                    = java-api
      org.jetbrains.kotlin.localToProject = public (not requested)
      org.jetbrains.kotlin.platform.type  = jvm (compatible with: androidJvm)
      org.gradle.status                   = release (not requested)
   ]

org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1
\--- androidReleaseCompileClasspath

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
   variant "jvm-api" [
      org.gradle.libraryelements         = jar (not requested)
      org.gradle.usage                   = java-api
      org.jetbrains.kotlin.platform.type = jvm (compatible with: androidJvm)
      org.gradle.status                  = release (not requested)
   ]
   Selection reasons:
      - By conflict resolution : between versions 1.4.1 and 1.3.9-native-mt-2

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
\--- org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1
     \--- androidReleaseCompileClasspath

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2 -> 1.4.1
\--- androidReleaseCompileClasspath

org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.4.1
   variant "jvm-api" [
      org.gradle.libraryelements         = jar (not requested)
      org.gradle.usage                   = java-api
      org.jetbrains.kotlin.platform.type = jvm (compatible with: androidJvm)
      org.gradle.status                  = release (not requested)
   ]

org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.4.1
\--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
     +--- androidReleaseCompileClasspath (requested org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2)
     \--- org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1
          \--- androidReleaseCompileClasspath
```
</details>

<details>
<summary>Mac OS X 64</summary>

```
./gradlew :core:dependencyInsight --configuration macosX64MainImplementationDependenciesMetadata --dependency coroutines

> Task :core:dependencyInsight
org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2 (forced)
   variant "metadata-api" [
      org.gradle.usage                   = kotlin-metadata
      org.jetbrains.kotlin.platform.type = common
      org.gradle.status                  = release (not requested)
   ]

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2
\--- macosX64MainImplementationDependenciesMetadata

org.jetbrains.kotlinx:kotlinx-coroutines-core:{strictly 1.3.9-native-mt-2} -> 1.3.9-native-mt-2
\--- macosX64MainImplementationDependenciesMetadata

org.jetbrains.kotlinx:kotlinx-coroutines-core-metadata:1.3.9-native-mt-2 (forced)
   variant "metadata-api" [
      org.gradle.usage                   = kotlin-metadata
      org.jetbrains.kotlin.platform.type = common
      org.gradle.status                  = release (not requested)
   ]

org.jetbrains.kotlinx:kotlinx-coroutines-core-metadata:1.3.9-native-mt-2
\--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2
     \--- macosX64MainImplementationDependenciesMetadata
```
</details>

<details>
<summary>JavaScript</summary>

```
./gradlew :core:dependencyInsight --configuration jsCompileClasspath --dependency coroutines

> Task :core:dependencyInsight
org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
   variant "jsLegacy-api" [
      org.gradle.usage                   = kotlin-api
      org.jetbrains.kotlin.js.compiler   = legacy
      org.jetbrains.kotlin.platform.type = js
      org.gradle.status                  = release (not requested)
   ]
   Selection reasons:
      - By conflict resolution : between versions 1.4.1 and 1.3.9-native-mt-2

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
\--- jsCompileClasspath

org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2 -> 1.4.1
\--- jsCompileClasspath

org.jetbrains.kotlinx:kotlinx-coroutines-core-js:1.4.1
   variant "jsLegacy-api" [
      org.gradle.usage                   = kotlin-api
      org.jetbrains.kotlin.js.compiler   = legacy
      org.jetbrains.kotlin.platform.type = js
      org.gradle.status                  = release (not requested)
   ]

org.jetbrains.kotlinx:kotlinx-coroutines-core-js:1.4.1
\--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.4.1
     \--- jsCompileClasspath (requested org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9-native-mt-2)
```
</details>

Reply:
The important bit is `!!` (strictly) on the Coroutines dependency for `macosX64`:

https://github.com/JuulLabs/kable/blob/4312acaec9fbc3baf3d352235a918fd692cf0c08/core/build.gradle.kts#L41-L46

Reply:
> Is this not considered a version conflict because `1.3.9-etc` is compatible with `1.4.1`?

To directly answer your original question: no, it's **not** because they are compatible. The versions just live completely independent on the platforms.

Reply:
> Dependency insights:
> 
> Android
> Mac OS X 64
> JavaScript

Today I learned about `<details><summary>...</summary>...</details>` and it's so cool.

Reply:
> Today I learned about `<details><summary>...</summary>...</details>` and it's so cool.

Right?! Just be sure to separate `<summary>..</summary>` with a [blank line before the actual details content](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab).